### PR TITLE
Use `display name` in forum rather than `login` for forum.

### LINKED ISSUE #946: Consider showing artist display names, along with or instead of /usernames/ in the forum

### PR DESCRIPTION
**Pull request by innatewonderer**
Fixes #946 
* [x] PR title accurately summarizes changes
Tested locally
alonetone___Welcome_to_your_messageboard_s_very_first_thread_and_jzueva_Ulybnis__ruby_

## Before code review *and after additional commits* during review.

* [x] Update title and description to account for additional changes
* [x] All tests green
* [x] Booted up the branch locally, exercised any new code
* [ ] Percy changes are purposeful or explained
* [ ] Css changes are happy on mobile (via Percy is ok)

### CODE REVIEW

**Comment by sudara** [comment]
Hey @innatewonderer! Thanks for tackling.

The reason that logins were chosen is because of the ability to @ someone on the forum, which is one of the key features we wanted. We'd need to:

1. Confirm that ability works with display names which can a wider range of characters (vs. logins which can't have spaces or special characters).
2. Confirm that the dropdown that appears when typing @ is populated with display names and NOT logins

If those two things are true, we can merge this. 

Otherwise the Plan A was to show the display name in the forum but *also* show the login with an @ in front of it, so people would know which username they could use to reply to someone. This would require overriding the thredded view (copying it into our app) where the user is displayed.

**Comment by innatewonderer** [comment]
Hmmmm does this look OK?
alonetone___Testing_new_topic

File: config/initializers/thredded.rb [hunk 1]
```diff
@@ -7,6 +7,10 @@
 # must be unique.
 Thredded.user_name_column = :login
 
+# User display name method, by default thredded uses the user_name_column defined above
+# You may want to use :to_s or some more elaborate method
+Thredded.user_display_name_method = :forum_name
```
Comment (innatewonderer) [inline config/initializers/thredded.rb]: I guess Thredded accounted for us cranky developers :)

**Comment by sudara** [comment]
@innatewonderer What shows up in the dropdown? Is it whatever `user_display_name_method` returns or is it `@username`

File: app/models/user.rb [hunk 2]
```diff
@@ -268,6 +268,10 @@ def avatar_image_location(variant:)
     )
   end
 
+  def forum_name
+    "@#{login} - #{display_name}"
```
Comment (sudara) [inline app/models/user.rb]:
```suggestion
    "#{display_name} (@#{login})"
```

**Comment by sudara** [comment]
Ahh, looking at the screenshot, it seems the UI uses both `user_display_name_method` and `login`? but only `login` for the mentions dropdown?

If so, then we are good to go, but I'd like to change the formatting to: `Display Name (@login)`

**Comment by innatewonderer** [comment]
> What shows up in the mentions dropdown? Is it whatever user_display_name_method returns or is it :login?

@sudara not sure why, but for me, there is no dropdown even on alonetone itself so I can't test it.

**Comment by innatewonderer** [comment]
Fullscreen_12_18_19__8_22_PM

**Comment by sudara** [comment]
I think we are missing a bit of clarity here — or at least I am, in part due to the `moderator Moderator(@moderator)` thing in the last screenshot — is that showing the login + display name (which then shows the login again?).

If we just have the forum_name be `:display_name`, can we search for the person using their display name? 

If so, that's ideal and adding the `(@login)` to the forum_name is probably overkill/confusing.

**Comment by innatewonderer** [comment]
Oh, hey, would ya look at that. We are able to show `display_name`

Fullscreen_1_31_20__6_39_PM

And I confirmed that it turns into a link when you click submit

Fullscreen_1_31_20__6_38_PM

The only issue is that it displays the name that contains a space in the middle as a string :/ a bit unappealing.

@sudara thoughts?

**Pull request merged**